static/js/src/advantage/offers/components/Offer/Offer.tsx

```diff
@@ -0,0 +1,58 @@
+import React from "react";
+import { ActionButton, Card } from "@canonical/react-components";
+import { currencyFormatter } from "advantage/react/utils";
+import { Offer as OfferType, Item } from "../../types";
+
+type Props = {
+  offer: OfferType;
+};
+
+const Offer = ({ offer }: Props) => {
+  return (
+    <Card data-testid="offer-card">
+      <div className="advantage-offer-row">
+        <p className="p-text--x-small-capitalised contract-item">
+          Contract item
+        </p>
+        <p className="p-text--x-small-capitalised allowance">Allowance</p>
+        <p className="p-text--x-small-capitalised price">Price</p>
+      </div>
+      <hr />
+      {offer.items.map((item: Item) => {
+        return (
+          <div key={item.id} className="advantage-offer-row">
+            <p className="contract-item">
+              <strong>{item.name}</strong>
+            </p>
+            <p className="allowance">{item.allowance ?? 0}</p>
+            <p className="price">
+              {currencyFormatter.format(item.price / 100)}
+            </p>
+          </div>
+        );
+      })}
+      <div className="advantage-offer-row">
+        <p className="p-text--x-small-capitalised allowance">
+          Total before taxes
+        </p>
+        <p className="price">{currencyFormatter.format(offer.total / 100)}</p>
+      </div>
+      <div
+        style={{ width: "100%", display: "flex", justifyContent: "flex-end" }}
+      >
+        <ActionButton
+          appearance="positive"
+          className="u-no-margin--bottom"
+          onClick={() => {
+            alert("nope.");
+          }}
+          disabled={!offer.actionable}
```

Comment:
Could you add a comment please. Seems strange to have an offer that is not actionable from the code.

Reply:
I'm not sure this is an actual use case but there is an `actionable` attribute so I figured out we should at least disable the button if it's not actionnable

Reply:
Would be interesting to find out why that is in the payload and add a note as this is likely to confuse the next person to touch this code

Reply:
@albertkol do you know if it comes from the CS API?

Reply:
We can filter them out. It's not a problem. I will check to see if we can fetch only actionable ones.

Reply:
Hiding unactionable offers could lead to confusion for users, depending on the reason why. If the user is expecting to see an offer and it expires it would be a better UX to display the offer with a message of why this is now unactionable.